daq/panda/test/testScript.py

```diff
@@ -52,17 +58,26 @@
   text = ser.readline().decode() #read in one data string
   print(milliseconds() - start)
 
+  #optional: Prints out black box data to certain location
+  save_path = 'C:\\Users\\jsand\\OneDrive\\Documents\\SMV\\DAQ\\Trials'
+  fileName = trialName + ".txt"
+  completeName = os.path.join(save_path, fileName)
+    
   #Setting the time for each trial
   now = datetime.now() #set time
   timeName = now.strftime("%H:%M:%S:%f") [:-3] #set current time
-
-  numValues = 22; #expected number of data inputs
-  j = 0 #iterating variable for traversing data string
+  text = text[:-2]
+  text = text + "." + timeName + "\r\n" #Something with newline in arduino is stopping this from actually being appended to string
+
+  #Printing inputs to Black Box
+  with open(completeName, 'a') as f:
+      print(text, file = f)
+ 
+  numValues = 22 #expected number of data inputs
+  data = text.split(";") #splits each data point into a list 
+  
   for i in range (0, numValues):
-    prefix = text[j:j+3] #KEY
-    data = text[j+4:j+6] #VALUE
-    inputs_dict[prefix] = float(data) #may fail on first run, simply re-run
-    j+=6 #assumes 2 digit data values
+        inputs_dict[prefix[i]] = float(data[i]) #may fail on first run, simply re-run
```

Comment:
i'm not convinced using a fix prefix array is the best approach, under the assumption that we might have sensor failures wherein we don't receive all 22 expected values when we want them

Reply:
Being removed!